src/instructlab/rag/process_docs.py

```diff
@@ -0,0 +1,156 @@
+# Standard
+from pathlib import Path
+from typing import Iterable
+import json
+import logging
+import os
+import tempfile
+import time
+
+# Third Party
+from docling.datamodel.base_models import ConversionStatus  # type: ignore
+from docling.datamodel.base_models import InputFormat  # type: ignore
+from docling.datamodel.document import ConversionResult  # type: ignore
+from docling.datamodel.pipeline_options import EasyOcrOptions  # type: ignore
+from docling.datamodel.pipeline_options import OcrOptions  # type: ignore
+from docling.datamodel.pipeline_options import PdfPipelineOptions  # type: ignore
+from docling.datamodel.pipeline_options import TesseractOcrOptions  # type: ignore
+from docling.document_converter import DocumentConverter  # type: ignore
+from docling.document_converter import PdfFormatOption  # type: ignore
+from docling.models.easyocr_model import EasyOcrModel  # type: ignore
+from docling.models.tesseract_ocr_model import TesseractOcrModel  # type: ignore
+
+# First Party
+from instructlab.rag.taxonomy_utils import lookup_knowledge_files
+from instructlab.utils import clear_directory
+
+logger = logging.getLogger(__name__)
+
+
+def process_docs_from_taxonomy(taxonomy_path, taxonomy_base, output_dir):
```

Comment:
I imagine that @cdoern will ask for more doc strings in all these functions (I agree tbh)